### PR TITLE
add app for getting api topics and error handling

### PR DESCRIPTION
**Pull request by karobarari**

### CODE REVIEW

File: app/app.controllers.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+const { selectTopics } = require('./app.models')
+
+
+exports.getAllTopics = ( req,res,next)=>{
+    selectTopics().then((topics)=>{
+        res.status(200).send(topics);
```
Comment (two-triangles) [inline app/app.controllers.js]: We should be sending the topics on a key that can be accessed - e.g { topics: fetchedTopics }

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const { getAllTopics } = require('./app.controllers')
+const { handleNotFoundError } = require('./errors')
+
+const app = express();
+app.use(express.json());
```
Comment (two-triangles) [inline app/app.js]: We do not need app.use(express.json( )) - we are not dealing with POST/PATCH requests yet.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+describe("GET /api/topics", () => {
+  test("API healthcheck ", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+  test("getting all topics", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual([
```
Comment (two-triangles) [inline __tests__/app.test.js]: There may be a point where a topic gets removed/gets added to the database. We should be testing in a more dynamic way. Make an assertion for how many topics are coming back and then make another assertion for the properties we expect there to be in each topic object in the database

File: app/errors.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+exports.handleNotFoundError = (err, req, res, next) => {
+  if (err.code === "42P01") {
+    res.status(404).send({ msg: err.msg });
+  } else next(err);
+};
```
Comment (two-triangles) [inline app/errors.js]: We should also have a 'handleServerError' error handler, which will act as a catch all for when we want to debug what our error is. This should log what the error is as well as returning a 500 error code and a 'server error' message.

**Review by nickkagame — APPROVED** [review]
Hi Karo, This is smart... You have done this in a more complicated way that necessary, but it makes sense, and looks like it'll work just fine.

**Pull request merged**